src/lib/UserEventHelper.ts

```diff
@@ -49,6 +66,46 @@ export async function getEventData(eid: string) {
   return data;
 }
 
+export async function verifyUserInEvent(eid: string, uid: string) {
+  if (!eid || !uid) {
+    throw new Error('Invalid event id, or invalid uid provided.');
+  }
+
+  const snapshot = await getDb()
+    .collection(DB_PATHS.EVENT_USERS)
+    .where('eid', '==', eid)
+    .where('uid', '==', uid)
+    .get();
+
+  if (snapshot.docs.length !== 1) {
+    throw new Error('Ticket does not exist.');
+  }
+
+  const doc = snapshot.docs[0];
+  const data = doc.data();
+
+  if (!doc.exists || !data) {
+    throw new Error('Ticket could not be found.');
+  }
+
+  return [doc, data];
+}
+
+export async function checkinUser(userEvent: any) {
+  if (!userEvent) {
```

Comment:
i don't think this condition will ever be true, cuz you already check the data before